dcrwallet.go

```diff
@@ -289,7 +290,22 @@ func rpcClientConnectLoop(legacyRPCServer *legacyrpc.Server, loader *wallet.Load
 				legacyRPCServer.SetChainServer(chainClient)
 			}
 			if cfg.EnableTicketBuyer {
-				startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				pm, err = startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				if err != nil {
+					log.Errorf("Unable to start ticket buyer: %v", err)
+					go func() {
+						loader.SetWaitFn(nil)
+					}()
+					return
+				}
+				pm.Start()
+				go func() {
```

Comment:
I think what I saw was `OpenWallet` grpc command hanging up. I think that's because it calls `OpenExistingWallet` with loader lock held, then calls `loader.onLoaded` and this callback calls `SetWaitFn` which again needs the same lock.